core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -709,11 +737,13 @@ class GroupMetadataManager(brokerId: Int,
     cleanupGroupMetadata(None)
   }
 
-  def cleanupGroupMetadata(deletedTopicPartitions: Option[Seq[TopicPartition]]) {
-    val startMs = time.milliseconds()
+  def cleanupGroupMetadata(deletedTopicPartitions: Option[Seq[TopicPartition]],
+                           groups: Iterable[GroupMetadata] = groupMetadataCache.values,
```

Comment:
It would be better not to have optional arguments. Let's make the caller provide the values explicitly.